Dispose extension files in EditorAppExtended

vrama628 opened the pull request:
Fixes #679 

There was a bug where disposing and then reinitializing an `EditorAppExtended` where an extension has `filesOrContents` would result in an exception being thrown due to not disposing these files from the last time the extension was registered. This fixes that by disposing of the results of `registerFileUrl` in the `disposeApp` method.

I added a test to exercise this; the test introduced in the first commit exhibits the `Error: file 'extension-file://undefined.undefined/' already exists` error, and that error goes away in the second commit. Unfortunately, when I run this test suite, I also get a bunch of "Array buffer allocation failed" errors that seem to come from somewhere in VSCode's services. I tried using the `updateExtendedAppPrototyp` test helper to fix this, but that seems to remove so much functionality that the original "file already exists" error no longer happens in the first place. Open to suggestions on how to fix the test, or I can just remove the test so we can ship the fix.

CGNonofr commented on packages/wrapper/src/editorAppExtended.ts:
```diff
@@ -106,6 +108,8 @@ export class EditorAppExtended extends EditorAppBase {
         this.disposeEditors();
         this.extensionRegisterResults.forEach((k) => k?.dispose());
+        this.subscriptions.forEach((k) => k.dispose());
+        this.subscriptions.length = 0;
```
nit: for informations (@kaisalmen ) `vcode/monaco` export the `DisposableCollection` class from VSCode, do not hesitate to use it :)

vrama628 replied:
Thanks, I've updated it to use `DisposableCollection`. I wasn't able to find it in the `vscode` or `monaco-editor` packages, but I did find it in the `vscode-ws-jsonrpc` package so I used it from there.

kaisalmen replied:
@CGNonofr you meant `import { DisposableStore } from 'vscode/monaco';` like used here https://github.com/CodinGame/monaco-vscode-api/blob/main/src/service-override/tools/views.ts#L258-L367, right?

vrama628 commented:
@kaisalmen Thanks, I've moved the test into a separate draft PR https://github.com/TypeFox/monaco-languageclient/pull/688 and updated to `import { DisposableStore } from 'vscode/monaco'`.

kaisalmen reviewed (approved):
LGTM. Thank you.
